src/account/Settings.tsx

```diff
@@ -411,12 +416,33 @@ export const Account = ({ navigation, route }: Props) => {
             <SettingsItemCta
               title={t('keylessBackupSettingsTitle')}
               onPress={
-                cloudBackupCompleted ? onPressDeleteKeylessBackup : onPressSetUpKeylessBackup
+                deleteKeylessBackupStatus === KeylessBackupDeleteStatus.InProgress
+                  ? () => {
+                      // do nothing
+                    }
+                  : cloudBackupCompleted
+                    ? onPressDeleteKeylessBackup
+                    : onPressSetUpKeylessBackup
               }
               testID="KeylessBackup"
-              ctaText={cloudBackupCompleted ? t('delete') : t('setup')}
-              ctaColor={cloudBackupCompleted ? colors.error : colors.primary}
-              showChevron={!cloudBackupCompleted}
+              cta={
+                <>
+                  {deleteKeylessBackupStatus === KeylessBackupDeleteStatus.InProgress && (
+                    <LoadingSpinner width={32} />
+                  )}
+                  <Text testID={`KeylessBackup/cta`} style={styles.value}>
+                    {deleteKeylessBackupStatus === KeylessBackupDeleteStatus.InProgress
+                      ? t('pleaseWait')
+                      : cloudBackupCompleted
+                        ? t('delete')
+                        : t('setup')}
+                  </Text>
+                  {!cloudBackupCompleted &&
+                    deleteKeylessBackupStatus !== KeylessBackupDeleteStatus.InProgress && (
+                      <ForwardChevron />
+                    )}
```

Comment:
this is quite hard to read with a nested ternaries. maybe we can render in progress and setup / delete as two subcomponents?

Reply:
+1 to this, seems like we could consolidate some of the  `deleteKeylessBackupStatus === KeylessBackupDeleteStatus.InProgress` checks and clean this up a bit?